src/test/lrsql/util/reaction_test.clj

```diff
@@ -90,3 +90,68 @@
            (-> simple-reaction-ruleset
                r/serialize-ruleset
                r/deserialize-ruleset)))))
+
+
+(def fake-cond-map
+  {:completed_a {"id"     "6fbd600f-d17c-4c74-801a-2ec2e53231f7"
+                  "actor"  {"mbox" "mailto:bob@example.com"
+                            "madeUpAttr" "8"}
+                  "verb"   {"id" "https://example.com/verbs/completed"}
+                  "object" {"id"         "https://example.com/activities/a"
+                            "objectType" "Activity"}
+                  "result" [{"success" true} {"success" false}]}
+   :completed_b   {"id"     "c51d1628-ae4a-449f-8d8d-13d57207f468"
+                    "actor"  {"mbox" "mailto:bob@example.com"}
+                    "verb"   {"id" "https://example.com/verbs/completed"}
+                    "object" {"id"         "https://example.com/activities/b"
+                              "objectType" "Activity"}
+                    "result" [{"success" false} {"success" true}]}})
+
+(deftest statement-generation-test
+  (testing "Basic statement generation"
+    (is (= (r/generate-statement
+            fake-cond-map
+            {"actor" {"mbox" {"$templatePath" [:completed_a "actor" "mbox"]}}
```

Comment:
Since the input for this will come from JSON, we won't be able to mix keywords and strings in the path like this. Your function should probably handle the first segment being a key so that the path could just be `["completed_a" "actor" "mbox"]`